tests/st/policy/test_felix_gateway.py

```diff
@@ -0,0 +1,510 @@
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'gw-ext',
+                'node': self.gateway_hostname,
+                'labels': {'role': 'gateway-ext'}
+            },
+            'spec': {
+                'interfaceName': 'eth1'
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    def add_host_iface(self):
+        host_endpoint_data = {
+            'apiVersion': 'v1',
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'host-int',
+                'node': self.host_hostname,
+                'labels': {'role': 'host'}
+            },
+            'spec': {
+                'interfaceName': 'eth0',
+                'expectedIPs': [self.get_container_ip('cali-st-host')],
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    @staticmethod
+    def _apply_resources(resources, host):
+        # Apply new resources.
+        host.writefile("resources",
+                       yaml.dump(resources, default_flow_style=False))
+        host.calicoctl("apply -f resources")
+
+    def assert_host_can_curl_ext(self):
+        try:
+            self.host.execute("curl --fail -m 1 -o /tmp/nginx-index.html %s" % self.ext_server_ip)
+        except subprocess.CalledProcessError:
+            _log.exception("Internal host failed to curl external server IP: %s",
+                           self.ext_server_ip)
+            self.fail("Internal host failed to curl external server IP: %s" % self.ext_server_ip)
+
+    def assert_host_can_not_curl_ext(self):
+        try:
+            self.host.execute("curl --fail -m 1 -o /tmp/nginx-index.html %s" % self.ext_server_ip)
+        except subprocess.CalledProcessError:
+            return
+        else:
+            self.fail("Internal host can curl external server IP: %s" % self.ext_server_ip)
+
+    def remove_pol_and_endpoints(self):
+        self.delete_all("pol")
+        self.delete_all("hostEndpoint")
+        # Wait for felix to remove the policy and allow traffic through the gateway.
+        retry_until_success(self.assert_host_can_curl_ext)
+
+    def delete_all(self, resource):
+        # Grab all objects of a resource type
+        objects = yaml.load(self.hosts[0].calicoctl("get %s -o yaml" % resource))
+        # and delete them (if there are any)
+        if len(objects) > 0:
+            self._delete_data(objects, self.hosts[0])
+
+    def _delete_data(self, data, host):
+        _log.debug("Deleting data with calicoctl: %s", data)
+        self._exec_calicoctl("delete", data, host)
+
+    @staticmethod
+    def _exec_calicoctl(action, data, host):
+        # use calicoctl with data
+        host.writefile("new_profile",
+                       yaml.dump(data, default_flow_style=False))
+        host.calicoctl("%s -f new_profile" % action)
+
+    @classmethod
+    def get_container_ip(cls, container_name):
```

Comment:
Can you use a DockerHost objects for your gateways?  This method looks much like something from `DockerHost.__init__()`